src/kayenta/edit/metricList.tsx

```diff
@@ -148,6 +156,14 @@ function mapDispatchToProps(dispatch: (action: Action & any) => void): IMetricLi
         }
       }));
     },
+    copyMetric: (metric: ICanaryMetricConfig) => {
+      const metricCopy: ICanaryMetricConfig = Object.assign({}, metric, {
```

Comment:
There are some deeply nested fields in a metric config - would use lodash `cloneDeep` here instead of a shallow copy.

Reply:
Good call, done.